### PR TITLE
[CI] Updated RBE Windows Image (MSVC 2022)

### PR DESCRIPTION
**Pull request by veblush**
Updated the Windows RBE image to use MSVC 2022

This is the last part of upgrading MSVC to 2022. This aligns with the minimum MSVC version required for gRPC, as stated in [the Foundational C++ Support Matrix](https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md).

An alternative & incremental approach to https://github.com/grpc/grpc/pull/37987
Follow-up of https://github.com/grpc/grpc/pull/38006

### CODE REVIEW

**Comment by veblush** [comment]
Bazel RBE Windows always fail with internal tool errors which are expected. This is a known issue of current RBE/Windows environment.

**Review by drfloob — APPROVED** [review]

File: third_party/toolchains/generate_windows_rbe_configs.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ docker pull ${WINDOWS_RBE_DOCKER_IMAGE}
 ${RBE_CONFIGS_GEN_TOOL_PATH} \
     --bazel_version="${BAZEL_VERSION}" \
     --toolchain_container="${WINDOWS_RBE_DOCKER_IMAGE}" \
-    --output_src_root="${REPO_ROOT}" \
+    --output_src_root=. \
```
Comment (drfloob) [inline third_party/toolchains/generate_windows_rbe_configs.sh:51]: Is this change intentional?

Comment (veblush) [reply third_party/toolchains/generate_windows_rbe_configs.sh:51]: Yep. Otherwise it doesn't work.
